diff --git a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/utils/CameraCharacteristicsUtils.kt b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/utils/CameraCharacteristicsUtils.kt
index 545505b..23e2861 100644
--- a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/utils/CameraCharacteristicsUtils.kt
+++ b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/utils/CameraCharacteristicsUtils.kt
@@ -55,5 +55,5 @@ fun CameraCharacteristics.getFieldOfView(): Double {
   val focalLengths = this.get(CameraCharacteristics.LENS_INFO_AVAILABLE_FOCAL_LENGTHS)!!
   val sensorSize = this.get(CameraCharacteristics.SENSOR_INFO_PHYSICAL_SIZE)!!
 
-  return 2 * atan(sensorSize.bigger / (focalLengths[0] * 2)) * (180 / PI)
+  return 2 * atan(sensorSize.height / (focalLengths[0] * 2)) * (180 / PI)
 }
